media-gfx/zbar/zbar-0.23.ebuild

```diff
@@ -0,0 +1,187 @@
+# Copyright 1999-2020 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=7
+
+# TODO: drop python2_7
+# * https://github.com/mchehab/zbar/issues/27#issuecomment-491463811
+# * https://bugs.gentoo.org/694800
+PYTHON_COMPAT=( python2_7 python3_{6,7,8} )
+
+inherit autotools flag-o-matic java-pkg-opt-2 multilib-minimal python-single-r1 virtualx
+
+DESCRIPTION="Library and tools for reading barcodes from images or video"
+HOMEPAGE="https://github.com/mchehab/zbar"
+SRC_URI="https://linuxtv.org/downloads/${PN}/${P}.tar.gz"
+
+KEYWORDS="~amd64 ~arm ~x86"
+LICENSE="LGPL-2.1"
+SLOT="0"
+
+IUSE="dbus graphicsmagick gtk gtk2 imagemagick introspection java jpeg png python qt5 static-libs test +threads v4l X xv"
+RESTRICT="!test? ( test )"
+REQUIRED_USE="
+	introspection? ( || ( gtk gtk2 ) )
+	python? ( ${PYTHON_REQUIRED_USE} )
+	test? ( ${PYTHON_REQUIRED_USE}
+		X png || ( graphicsmagick imagemagick ) )
+	qt5? ( X )"
+
+COMMON_DEPEND="
+	dbus? ( sys-apps/dbus[${MULTILIB_USEDEP}] )
+	gtk? (
+		dev-libs/glib:2[${MULTILIB_USEDEP}]
+		x11-libs/gtk+:3[${MULTILIB_USEDEP}] )
+	gtk2? (
+		dev-libs/glib:2[${MULTILIB_USEDEP}]
+		x11-libs/gtk+:2[${MULTILIB_USEDEP}] )
+	graphicsmagick? ( media-gfx/graphicsmagick:= )
+	imagemagick? ( media-gfx/imagemagick:= )
+	introspection? ( dev-libs/gobject-introspection )
+	jpeg? ( virtual/jpeg:0[${MULTILIB_USEDEP}] )
+	png? (
+		graphicsmagick? ( media-gfx/graphicsmagick:=[png] )
+		imagemagick? ( media-gfx/imagemagick:=[png] ) )
+	python? (
+		${PYTHON_DEPS}
+		gtk2? ( $(python_gen_cond_dep 'dev-python/pygtk:2[${PYTHON_USEDEP}]' python2_7) ) )
+	qt5? (
+		dev-qt/qtcore:5
+		dev-qt/qtgui:5
+		dev-qt/qtwidgets:5 )
+	v4l? ( media-libs/libv4l:0=[${MULTILIB_USEDEP}] )
+	X? (
+		qt5? ( dev-qt/qtx11extras:5 )
+		x11-libs/libX11[${MULTILIB_USEDEP}]
+		x11-libs/libXext[${MULTILIB_USEDEP}]
+		xv? ( x11-libs/libXv[${MULTILIB_USEDEP}] ) )"
+
+RDEPEND="${COMMON_DEPEND}
+	java? ( virtual/jre:= )"
+
+DEPEND="${COMMON_DEPEND}
+	java? (
+		>=virtual/jdk-1.8:=
+		test? (
+			dev-java/junit:4
+			dev-java/hamcrest-core:1.3 ) )
+	test? (
+		${PYTHON_DEPS}
+		dev-python/pillow[${PYTHON_USEDEP}] )"
+
+BDEPEND="
+	gtk? ( dev-util/glib-utils )
+	gtk2? ( dev-util/glib-utils )
+	sys-devel/gettext
+	virtual/pkgconfig"
+
+pkg_setup() {
+	if use python || use test; then
+		python-single-r1_pkg_setup
+	fi
+	java-pkg-opt-2_pkg_setup
+}
+
+src_prepare() {
+	eapply "${FILESDIR}/${PN}-0.10-errors.patch"
+
+	# fix typo
+	# * https://github.com/mchehab/zbar/issues/73
+	eapply "${FILESDIR}/${P}-fix_python_detect.patch"
```

Comment:
Please move those into global PATCHES array, declare it below the BDEPEND block.